server/accounts.go

```diff
@@ -703,12 +743,22 @@ func (a *Account) selectMappedSubject(dest string) (string, bool) {
 	var d *destination
 	var ndest string
 
+	dests := m.dests
+	if len(m.cdests) > 0 {
+		cn := a.srv.ClusterName()
```

Comment:
Lock hierarchie violation?
We hold a.mu, will lock s.mu
```
// clusterName returns our cluster name which could be dynamic.
func (s *Server) ClusterName() string {
	s.mu.Lock()
	cn := s.info.Cluster
	s.mu.Unlock()
	return cn
}
```

Reply:
It was that or unlock which @kozlovic is not a fan of.. And I can't hold it since its now dynamic (even without reload).

Reply:
Could create its own lock..

Reply:
ok fixed.